tasks/download.sh

```diff
@@ -35,8 +35,19 @@ verify-file() {
 }
 
 download() {
+  local tmp_file_name="pe-tmp-file"
+
   printf '%s\n' "Downloading: ${1}"
-  curl -s -f -L -o "$2" "$1"
+  tmp_file=$(mktemp -p /tmp/ "$tmp_file_name")
```

Comment:
The `-p` flag isn't supported by all `mktemp` implementations. Plus, we don't know for sure that `/tmp` is actually where temp files should go on any given platform. Better to let `mktemp` make those decisions itself.

```suggestion
  tmp_file=$(mktemp "peadm-download")
```